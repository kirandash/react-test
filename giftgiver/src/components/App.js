import React from 'react';
import { Button } from 'react-bootstrap';
import Gift from './Gift';
import { max_number } from '../helper'

class App extends React.Component {
    constructor() {
        super();
        this.state = { gifts: [] };
    }

    addGift = () => {
        // local copy of state variable
        const { gifts } = this.state;
        // const ids = this.state.gifts.map(gift => gift.id);
        // const max_id = max_number(ids);
        // possible branching needs to be tested as well - thus: create separate utility fn
        // const max_id = ids.length > 0 ? Math.max(...ids) : 0;
        gifts.push({ id: max_number(this.state.gifts.map(gift => gift.id)) + 1 })
        this.setState({ gifts });
    }

    removeGift = id => {
        const gifts = this.state.gifts.filter(gift => gift.id !== id);

        this.setState({ gifts });
    }

    render() {
        return (
            <div>
                <h2>Gift Giver</h2>
                <div className="gift-list">
                    {
                        this.state.gifts.map(gift => {
                            return (
                                <Gift 
                                    key={gift.id}
                                    gift={gift}
                                    removeGift={this.removeGift}
                                />
                            )
                        })
                    }
                </div>
                <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
            </div>
        )
    }
}

export default App;
