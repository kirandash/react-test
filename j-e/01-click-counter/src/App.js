import React, {Component} from 'react';
import './App.css';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      counter: 0
    }
  }

  increment = () => {
    this.setState({ counter: this.state.counter + 1 })
  }

  render() {
    return (
    <div data-test="component-app">
      <h1 data-test="counter-display">{this.state.counter}</h1>
      <button data-test="increment-button" onClick={this.increment}>Increment counter</button>
    </div>
  )};
}

export default App;
