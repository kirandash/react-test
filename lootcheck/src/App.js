import './App.css';
import Wallet from './components/Wallet';
import Loot from './components/Loot';

function App() {
  return (
    <div className="App">
      <h2>Loot Check</h2>
      <hr/>
      <Wallet/>
      <hr/>
      <Loot />
    </div>
  );
}

export default App;
