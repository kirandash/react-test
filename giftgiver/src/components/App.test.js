import React from 'react';
// helps in creating a shallow DOM
import { shallow } from 'enzyme';
import App from './App';

const app = shallow(<App />);

// Snapshot Test
it('renders correctly', () => {
    // Note: app.debug() works (app) does not
    expect(app.debug()).toMatchSnapshot();
});

// UI - unit test - for state variables initialization
it('initializes the `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
});

it('adds a new gift to `state` when clicking the `add gift` button', () => {
    app.find('.btn-add').simulate('click');
    expect(app.state().gifts).toEqual([{ id: 1 }]);
});
