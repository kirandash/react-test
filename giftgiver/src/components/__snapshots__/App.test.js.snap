// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div>
  <h2>
    Gift Giver
  </h2>
  <Button className=\\"btn-add\\" onClick={[Function (anonymous)]} variant=\\"primary\\" active={false} disabled={false}>
    Add Gift
  </Button>
</div>"
`;
